docker/Makefile

```diff
@@ -0,0 +1,51 @@
+git_describe := $(shell git describe)
+vcs_ref := $(shell git rev-parse HEAD)
+build_date := $(shell date -u +%FT%T)
+hadolint_available := $(shell hadolint --help > /dev/null 2>&1; echo $$?)
+hadolint_command := hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001
+hadolint_container := hadolint/hadolint:latest
+base_image := ubuntu:16.04
+
+ifeq ($(IS_NIGHTLY),true)
+	dockerfile := Dockerfile.nightly
+	version := puppet6-nightly
+else
+	version := $(shell echo $(git_describe) | sed 's/-.*//')
+	dockerfile := Dockerfile
+endif
+
+
+prep:
+ifeq ($(IS_NIGHTLY),true)
+	@echo "No prep needed!"
+else
+	@git pull --unshallow > /dev/null 2>&1 ||:
+	@git fetch origin 'refs/tags/*:refs/tags/*' > /dev/null 2>&1
+endif
+
+lint:
+ifeq ($(hadolint_available),0)
+	@$(hadolint_command) r10k/$(dockerfile)
+else
+	@docker pull $(hadolint_container)
+	@docker run --rm -v $(PWD)/r10k/$(dockerfile):/Dockerfile -i $(hadolint_container) $(hadolint_command) Dockerfile
+endif
+
+build: prep
+	@docker pull $(base_image)
```

Comment:
Do we really need this?

Reply:
yeah, otherwise we won't get updates when new ubuntu containers are published since we have shared build hardware